pkg/apis/planetscale/v2/vitessshard_methods.go

```diff
@@ -151,3 +153,47 @@ func (s *VitessShardSpec) CellInCluster(cellName string) bool {
 	_, inZoneMap := s.ZoneMap[cellName]
 	return inZoneMap
 }
+
+// NewVitessShardCondition returns a VitessShardCondition object based on the provided type and initial state.
+func NewVitessShardCondition(ty VitessShardConditionType, initState corev1.ConditionStatus) *VitessShardCondition {
+	now := v1.NewTime(time.Now())
+	return &VitessShardCondition{
+		Type:               ty,
+		Status:             initState,
```

Comment:
I think we should always make Status start as Unknown, and Reason and Message should be empty. That's the idiomatic "zero value" of a condition struct. This also would encourage all updates to go through ChangeStatus() rather than calling New and passing in a status but not a reason nor message.

Reply:
Good suggestion. Changed: https://github.com/planetscale/vitess-operator/pull/31/commits/19a40e6a420384f2e9fe11fe824dc076054f8df1.